src/openforms/payments/views.py

```diff
@@ -0,0 +1,306 @@
+    register = register
+
+
+@extend_schema(
+    summary=_("Start payment flow"),
+    description=_(
+        "This endpoint provides information to start external login flow for a submission."
+        "\n\nDue to support for legacy platforms this view doesn't redirect but provides information for the frontend to be used client side."
+        "\n\nVarious validations are performed:"
+        "\n* the form and submission must require payment"
+        "\n* the `plugin_id` is configured on the form"
+        "\n* payment is required and configured on the form"
+        "\n* the `next` parameter must be present"
+        "\n* the `next` parameter must match the CORS policy"
+    ),
+    parameters=[
+        OpenApiParameter(
+            name="uuid",
+            location=OpenApiParameter.PATH,
+            type=OpenApiTypes.STR,
+            description=_("UUID identifying the submission."),
+        ),
+        OpenApiParameter(
+            name="plugin_id",
+            location=OpenApiParameter.PATH,
+            type=OpenApiTypes.STR,
+            description=_(
+                "Identifier of the payment plugin. Note that this is validated "
+                "against the configured available plugins for this particular form."
+            ),
+        ),
+        OpenApiParameter(
+            name="next",
+            location=OpenApiParameter.QUERY,
+            type=OpenApiTypes.URI,
+            description=_(
+                "URL of the form to redirect back to. This URL is validated "
+                "against the CORS configuration."
+            ),
+            required=True,
+        ),
+    ],
+    responses={
+        (200, "application/json"): OpenApiResponse(
+            response=PaymentInfoSerializer,
+            description=_(
+                "Depending on the 'type' send a GET or POST request with the 'data' as Form Data to given 'url'."
+            ),
+        ),
+        (400, "text/html"): OpenApiResponse(
+            response=str, description=_("Bad request. Invalid parameters were passed.")
+        ),
+        (404, "text/html"): OpenApiResponse(
+            response=str,
+            description=_("Not found. The slug did not point to a live submission."),
+        ),
+    },
+)
+class PaymentStartView(PaymentFlowBaseView, GenericAPIView):
+    lookup_field = "uuid"
+    lookup_url_kwarg = "uuid"
+    queryset = Submission.objects.all()
+    serializer_class = PaymentInfoSerializer
+
+    def post(self, request, uuid: str, plugin_id: str):
+        submission = self.get_object()
+        try:
+            plugin = self.register[plugin_id]
+        except KeyError:
+            return HttpResponseBadRequest("unknown plugin")
+
+        if not submission.form.payment_required:
+            return HttpResponseBadRequest("payment not required")
+
+        if plugin_id != submission.form.payment_backend:
+            return HttpResponseBadRequest("plugin not allowed")
+
+        form_url = request.GET.get("next")
```

Comment:
```suggestion
        form_url = request.query_params.get("next")
```
You're in the `def post` handler trying to access `.GET` - it'll work, but it's weird. DRF request objects have a `query_params` attribute to deal with this.